source/RoslynAnalyzers/MessageContractAnalyzers_Conventions.cs

```diff
@@ -104,6 +104,11 @@ static bool PropertiesOnMessageTypes_MustHaveAtLeastOneValidationAttribute(Synta
 
         static bool MessageTypes_MustHaveXmlDocComments(SyntaxNodeAnalysisContext context, TypeDeclarationSyntax typeDec)
         {
+            // Filter out types that have [Experimental] attribute, those don't need XMLDocs.
+            // The syntax model is faster than the semantic model so we check this first 
+            var attrNames = typeDec.AttributeLists.SelectMany(al => al.Attributes.Select(a => (a.Name as IdentifierNameSyntax)?.Identifier.Text));
+            if (attrNames.Any(a => a == "Experimental")) return true;
```

Comment:
I would've expected this be `ExperimentalAttribute`. 
Hmm. I see your test - doesn't appear to need to be. Odd 🤷‍♂️

Reply:
Yeah, this is done at the syntax level; mostly runs of text with some hierarchy and loose typing (which is why it's faster), but it's not a perfect match for the object model in the compiler.